src/sonic-framework/rebootbackend/rebootbe.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+#include "dbconnector.h"
+#include "notificationconsumer.h"
+#include "notificationproducer.h"
+#include "reboot_common.h"
+#include "reboot_interfaces.h"
+#include "reboot_thread.h"
+#include "selectableevent.h"
+#include "status_code_util.h"
+
+namespace rebootbackend {
+
+#define REBOOT_REQUEST_NOTIFICATION_CHANNEL "Reboot_Request_Channel"
+#define REBOOT_RESPONSE_NOTIFICATION_CHANNEL "Reboot_Response_Channel"
+#define REBOOT_KEY "Reboot"
+#define REBOOT_STATUS_KEY "RebootStatus"
+#define CANCEL_REBOOT_KEY "CancelReboot"
+#define DATA_TUPLE_KEY "MESSAGE"
+
+class RebootBE {
+ public:
+  struct NotificationRequest {
+    std::string op;
+    std::string ret_string;
+  };
+
+  enum class RebManagerStatus {
+    WARM_INIT_WAIT,
+    IDLE,
+    COLD_REBOOT_IN_PROGRESS,
+    WARM_REBOOT_IN_PROGRESS
+  };
+
+  RebootBE(DbusInterface &interface);
+
+  RebManagerStatus GetCurrentStatus();
+
+  void Start();
+  void Stop();
+
+ private:
+  std::mutex m_status_mutex;
+  RebManagerStatus m_current_status = RebManagerStatus::IDLE;
```

Comment:
honest question, what does the prefix `m_` refers to?

Reply:
Seems pretty widely used sonic-swss-common: table.h, ipprefix.h, select.h
Core SONiC functionality.

Here’s a reddit discussion: https://www.reddit.com/r/csharp/comments/950s3l/prefixing_private_fields_with_m/

Seems like a good quote: “It's not new. It's part of Hungarian notation. That was MS official naming convention before .NET when we were still writing C++. It's not recommented anymore for C#.”

In same reddit some people think it means: member variable.